pkg/cli/admin/top/persistentvolumeclaims.go

```diff
@@ -0,0 +1,382 @@
+		}
+	}
+
+	if o.allNamespaces && len(args) != 0 {
+		return fmt.Errorf("a persistentvolumeclaim resource cannot be retrieved by name across all namespaces.")
+	}
+
+	return nil
+}
+
+type persistentVolumeClaimInfo struct {
+	Namespace       string
+	Name            string
+	UsagePercentage string
+}
+
+func (v persistentVolumeClaimInfo) PrintLine(out io.Writer) {
+	printValue(out, v.Namespace)
+	printValue(out, v.Name)
+	printValue(out, v.UsagePercentage)
+}
+
+func (o *options) Run(ctx context.Context, args []string) error {
+	o.BearerToken = o.RESTConfig.BearerToken
+	if len(o.RESTConfig.BearerToken) == 0 {
+		klog.V(4).Info("no token is currently in use for this session")
+		klog.V(5).Info("attempting to retrieve token from secret \"localhost-recovery-client-token\" in namespace \"openshift-kube-apiserver\"")
+		secret, err := o.ClientSet.CoreV1().Secrets("openshift-kube-apiserver").Get(context.TODO(), "localhost-recovery-client-token", metav1.GetOptions{})
+		if err != nil {
+			klog.V(4).Info(fmt.Errorf("error retrieving secret: %s", err.Error()))
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		localhostRecoveryToken, exist := secret.Data["token"]
+		if !exist {
+			klog.V(4).Info(fmt.Errorf("\"token\" key not found in secret \"localhost-recovery-client-token\" in namespace \"openshift-kube-apiserver\""))
+			return fmt.Errorf("no token is currently in use for this session")
+		}
+		o.BearerToken = string(localhostRecoveryToken)
+	}
+	persistentVolumeClaimsBytes, err := GetPersistentVolumeClaims(ctx, o.getRoute, o.BearerToken, o.Namespace, o.InsecureTLS, args)
+	if err != nil {
+		return err
+	}
+	promOutput := &PromOutput{}
+	err = json.Unmarshal([]byte(persistentVolumeClaimsBytes), &promOutput)
+	if err != nil {
+		return err
+	}
+
+	if len(promOutput.Data.Result) == 0 {
+		if o.Namespace == "" {
+			return fmt.Errorf("no persistentvolumeclaims found.")
+		} else {
+			if len(args) == 0 {
+				return fmt.Errorf("no persistentvolumeclaims found in %s namespace.", o.Namespace)
+			} else {
+				return fmt.Errorf("persistentvolumeclaim %q not found in %s namespace.", args[0], o.Namespace)
+			}
+		}
+	}
+
+	// if more pvc are requested as args but one of them does not not exist
+	if len(args) != 0 && len(promOutput.Data.Result) != len(args) {
+		resultingPvc := make(map[string]bool)
+		for _, _promOutputDataResult := range promOutput.Data.Result {
+			pvcName := _promOutputDataResult.Metric["persistentvolumeclaim"]
+			resultingPvc[pvcName] = true
+		}
+		for _, arg := range args {
+			_, pvcPresentInResult := resultingPvc[arg]
+			if !pvcPresentInResult {
+				return fmt.Errorf("persistentvolumeclaim %q not found in %s namespace.", arg, o.Namespace)
+			}
+		}
+
+	}
+	headers := []string{"NAMESPACE", "NAME", "USAGE(%)"}
+	pvcInfos := []persistentVolumeClaimInfo{}
+	infos := []Info{}
+	for _, _promOutputDataResult := range promOutput.Data.Result {
+		namespaceName := _promOutputDataResult.Metric["namespace"]
```

Comment:
How do we ensure that `namespace` key exists and this won't panic?

Reply:
I'm sure about that as `PersistentVolumeClaims` are namespaced resources and for all namespaces resources that field must be there.